signer/src/transaction_coordinator.rs

```diff
@@ -310,15 +309,7 @@ where
         let should_coordinate_dkg =
             should_coordinate_dkg(&self.context, &bitcoin_chain_tip).await?;
         let aggregate_key = if should_coordinate_dkg {
-            let dkg_result = self.coordinate_dkg(&bitcoin_chain_tip).await?;
-            // TODO: in `run_dkg_from_scratch` test, `dkg_result` differs from
-            // value fetched from the db. Adding a temporary fix for the (probably)
-            // race condition, but we should address this properly.
-            self.get_signer_set_and_aggregate_key(&bitcoin_chain_tip)
-                .await
-                .ok()
-                .and_then(|res| res.0)
-                .unwrap_or(dkg_result)
+            self.coordinate_dkg(&bitcoin_chain_tip).await?
```

Comment:
Assuming this goes well, shouldn't we refresh the state? It should have a new aggregate key now

Reply:
Do we want to do that before we see a confirmed rotate-keys event? (which would then be at a new bitcoin block and thus the state be updated)

Reply:
I guess this is only relevant on the first DKG, where after `coordinate_dkg` we would have an aggregate key while before we wouldn't, so going from picking the bootstrapping set to the latest dkg entry.

Reply:
Only applies to bootstrapping, and updating here wouldn't be enough because also the signers would need to refresh the state. The extra complexity doesn't seem to be worthy, resolving -- worst case, we are gonna "pay" this with an extra block latency before submitting the rotate tx